daemon/endpoint.go

```diff
@@ -46,41 +47,42 @@ func (h *getEndpoint) Handle(params GetEndpointParams) middleware.Responder {
 	log.WithField(logfields.Params, logfields.Repr(params)).Debug("GET /endpoint request")
 
 	var wg sync.WaitGroup
+	var epListLock lock.Mutex
 
 	if params.Labels == nil {
-		i := 0
 		endpointmanager.Mutex.RLock()
-		eps := make([]*models.Endpoint, len(endpointmanager.Endpoints))
+		eps := make([]*models.Endpoint, 0, len(endpointmanager.Endpoints))
 		wg.Add(len(endpointmanager.Endpoints))
-		for k := range endpointmanager.Endpoints {
-			go func(wg *sync.WaitGroup, i int, ep *endpoint.Endpoint) {
-				eps[i] = ep.GetModel()
+		for _, v := range endpointmanager.Endpoints {
+			go func(wg *sync.WaitGroup, ep *endpoint.Endpoint) {
+				epListLock.Lock()
+				eps = append(eps, ep.GetModelRLocked())
```

Comment:
How does taking `epListLock` relief you from taking the endpoint's individual lock?

Reply:
Good catch.